pageserver/src/pgdatadir_mapping.rs

```diff
@@ -388,6 +387,21 @@ impl<R: Repository> DatadirTimeline<R> {
 
         Ok(result.to_keyspace())
     }
+
+    pub fn repartition(&self, lsn: Lsn) -> Result<(KeyPartitioning, Lsn)> {
+        let partitioning_guard = self.partitioning.read().unwrap();
```

Comment:
Let's change it from RwLock to Mutex. Makes it a little bit simpler. There is only this one place where the lock is used, and this is only called from the compaction thread, so there is no need to allow concurrent readers, and this isn't performance critical.

Reply:
Done